multicluster/controllers/multicluster/memberclusterannounce_controller.go

```diff
@@ -256,61 +185,45 @@ func (r *MemberClusterAnnounceReconciler) processMCSStatus() {
 	}
 }
 
-/******************************* MemberClusterStatusManager methods *******************************/
-
-func (r *MemberClusterAnnounceReconciler) AddMember(memberID common.ClusterID) {
+func (r *MemberClusterAnnounceReconciler) addOrUpdateMemberStatus(memberID common.ClusterID) {
 	r.mapLock.Lock()
 	defer r.mapLock.Unlock()
-	if _, ok := r.memberStatus[memberID]; ok {
-		// already present
+	if data, ok := r.memberStatusMap[memberID]; ok {
+		klog.V(2).InfoS("Reset lastUpdateTime", "cluster", memberID)
+		// Reset lastUpdateTime for this member.
+		data.lastUpdateTime = time.Now()
+		for i, c := range data.status.Conditions {
+			if c.Type == multiclusterv1alpha1.ClusterConnected && data.status.Conditions[i].Reason != ReasonConnected {
+				data.status.Conditions[i].Message = "Member Connected"
+				data.status.Conditions[i].Reason = ReasonConnected
+			}
+		}
 		return
 	}
-
-	conditions := make([]multiclusterv1alpha1.ClusterCondition, 0, 2)
+	conditions := make([]multiclusterv1alpha1.ClusterCondition, 0, 1)
 	conditions = append(conditions, multiclusterv1alpha1.ClusterCondition{
 		Type:               multiclusterv1alpha1.ClusterReady,
 		Status:             v1.ConditionUnknown,
 		LastTransitionTime: metav1.Now(),
-		Message:            "Member created",
-		Reason:             ReasonNeverConnected,
-	})
-	conditions = append(conditions, multiclusterv1alpha1.ClusterCondition{
-		Type:               multiclusterv1alpha1.ClusterConnected,
-		Status:             v1.ConditionFalse,
-		LastTransitionTime: metav1.Now(),
-		Message:            "Member created",
-		Reason:             ReasonNeverConnected,
+		Message:            "Member Created",
```

Comment:
Question - what is difference between  "Member Created" and "Member Connected"? Should we just set the condition to "ConditionTrue + Member Connected"? Or there is a protocol that member should notify leader after it is ready by updating the Announce?

Reply:
`"Member Created"` will be used only when controller starts.  It will become `Member Connected"` when member cluster send a MemberClusterAnnounce update. The purpose is handle the case `MemberClusterAnnounce` aleady exists on leader cluster, but it might be stale and will be cleaned up soon.

Reply:
But I meant what is the problem we set the condition to "ConditionTrue + Member Connected" directly here? If the member cluster does not update the Announce, it will be changed to "Disconnected" anyway?

Reply:
Yes, it will be changed to "Disconnected", I feel both way works. Do you recommend to use `Member Connected` only?

Reply:
It will be updated every 5 seconds if member cluster works well. But you are right, I feel stale MemberClusterAnnounce case will be rare, let me change it to "ConditionTrue + Member Connected".

Reply:
BTW, do you think 5 seconds may be too frequent?

Reply:
Yeah, I feel  it can be longer, how about 15s?

Reply:
Was thinking about 10 seconds. 15 seconds sound good to me too.

Reply:
OK, I updated it to 10s in this PR, it can be quicker to detect failure.